### PR TITLE
initial set of unit tests

### PR DESCRIPTION
**Pull request by LuisFdF**
* custom homo/heterozygous freq threshold
* unit tests using pytest

### CODE REVIEW

File: t/test_tgpolyt_depths.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    :return:
+    '''
+
+    logger.info('testing count_tgpolyt()')
+
+    tgpolyt_counts = count_tgpolyt(get_reads_covering_region( bam_file, REGION), REGION)
+
+    if not ('(TG)11-7T', 16) in tgpolyt_counts[0].items():
+        logger.error('could not find variant (TG)11-7T with 16 reads: {}'.format(tgpolyt_counts))
+        assert 0
```
Comment (ah19) [inline t/test_tgpolyt_depths.py]: i think you can just do `assert ('(TG)11-7T', 16) in tgpolyt_counts[0].items()`

File: t/test_tgpolyt_depths.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    assert hethom_call(['11-9T', '5-2T'], [float(4) / 20, float(4) / 20]) == '--/--'
+
+    assert hethom_call(['11-9T', '11-7T', '5-2T'], [float(10) / 30, float(10) / 30, float(10) / 30]) == 'het/het/het'
+
+    assert hethom_call(['11-9T', '5-2T'], [float(16) / 20, float(4) / 20], hom_freq_threshold = 0.75) == 'hom alt/--'
+
+    assert hethom_call(['11-9T', '5-2T'], [float(7) / 20, float(4) / 20], hom_freq_threshold =  0.35) == 'het/--'
+
+    pass
```
Comment (ah19) [inline t/test_tgpolyt_depths.py]: if there is no newline at the end of the file, whenever you change the last line of the file it comes up as this line changing.

there's a setting in pycharm i added to make it always add a newline:
https://stackoverflow.com/questions/36043061/automatically-add-newline-on-save-in-pycharm

you also don't need a pass here

File: t/test_tgpolyt_depths.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+def test_hethom_call():
+    '''
+    testing different scenarios with hethom_call
+
+    '''
+    logger.info('testing hethom_call()')
+
+    assert hethom_call(['11-7T'], [float(16) / 17]) == 'hom'
```
Comment (ah19) [inline t/test_tgpolyt_depths.py]: if you add `from __future__ import division` to the top of the file you can get rid of all these float conversions:

```
In [1]: 1 / 2
Out[1]: 0

In [2]: from __future__ import division

In [3]: 1 / 2
Out[3]: 0.5
```

File: t/test_tgpolyt_depths.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    assert variant == "(TG)3-4T"
+
+    # precessing TG but not followed by polyT
+    variant = count_tg_t("ACTTGAATGTGTTTTTGTG")
+    if variant != "(TG)2-5T":
+        logger.warning('missing variant (TG)2-5T {}'.format(variant))
+    assert variant == "(TG)1-0T"
+
+
+    pass
```
Comment (ah19) [inline t/test_tgpolyt_depths.py]: this should be removed i think?

File: t/test_tgpolyt_depths.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    logger.info('testing get_reads_covering_region()')
+    #count reads in specified region
+    reads = get_reads_covering_region( bam_file, REGION)
+    assert len(reads) == 21
+
+    #count reads in region without reads
+    reads = get_reads_covering_region( bam_file, ['8', 117188618, 117188756, 'TG-polyT'])
+    assert len(reads) == 0
+
+    pass
```
Comment (ah19) [inline t/test_tgpolyt_depths.py]: no need for pass here

File: t/test_tgpolyt_depths.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    '''
+
+    logger.info('testing count_tgpolyt()')
+
+    tgpolyt_counts = count_tgpolyt(get_reads_covering_region( bam_file, REGION), REGION)
+
+    if not ('(TG)11-7T', 16) in tgpolyt_counts[0].items():
+        logger.error('could not find variant (TG)11-7T with 16 reads: {}'.format(tgpolyt_counts))
+        assert 0
+    pass
```
Comment (ah19) [inline t/test_tgpolyt_depths.py]: pass not required

File: t/test_tgpolyt_depths.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    assert get_tgpolyt_bases(read,REGION[1], REGION[2]) == 'ATGTGTGTGTGTGTGTGTGTGTGTTTTTTTA'
+
+    read = read_dict['HISEQ1:17:H947YADXX:2:1208:7897:88820']
+    assert get_tgpolyt_bases(read, REGION[1], REGION[2]) == ''
+
+    # sequence that possibly causes an issue when calling polyT
+    read = read_dict['HISEQ1:17:H947YADXX:1:2104:17965:28539']
+    assert get_tgpolyt_bases(read, REGION[1], REGION[2]) == 'ATGTGTGTGTGTTTGTGTGTGTGTTTTTTTA'
+
+    pass
```
Comment (ah19) [inline t/test_tgpolyt_depths.py]: not needed

**Review by ah19 — CHANGES_REQUESTED** [review]

**Comment by LuisFdF** [comment]
changes made

**Review by ah19 — APPROVED** [review]

**Pull request merged**